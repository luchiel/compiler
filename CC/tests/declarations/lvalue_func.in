int a;
int* f() { return &a; }

void main() { *f() = 5; }