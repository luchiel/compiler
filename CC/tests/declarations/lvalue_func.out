type <int>
type <double>
type <void>
type <int*>
	pointer to <int>
type <void*>
	pointer to <void>
variable NULL of type <void*> initialized by
	{int 0}
		type <int>
function printf
	returns type <int>
	arguments
		<no args>
variable a of type <int>
function f
	returns pointer to <int>
	arguments
		<no args>
	{{}}
	| 
	+-{return}
	  | 
	  +-{&}
					pointer to <int>
	    | 
	    +-{a}
					type <int>
function main
	returns type <void>
	arguments
		<no args>
	{{}}
	| 
	+-{;}
	  | 
	  +-{=}
				type <int>
	    | 
	    +-{*}
					type <int>
	    | | 
	    | +-{()}
							pointer to <int>
	    |   | 
	    |   +-{f}
							function f
								returns pointer to <int>
								arguments
									<no args>
	    | 
	    +-{int 5}
					type <int>
